From bfe98c83bdcc0f3c33d836e03e6d4cf7199c8d37 Mon Sep 17 00:00:00 2001
From: Jan Tojnar <jtojnar@gmail.com>
Date: Thu, 13 Oct 2022 10:16:32 +0200
Subject: [PATCH] build: Allow using V8 from system

Building V8 is slow and pointless on systems that already provide a recent version like some Linux distros.
With this change, you can build against system V8 with the following:

    make USE_SYSTEM_V8=1 SHLIB_LINK=-lv8 V8_OUTDIR=/usr/lib
---
 Makefile | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/Makefile b/Makefile
index 1180fae..c74b23d 100644
--- a/Makefile
+++ b/Makefile
@@ -13,6 +13,7 @@ OBJS = $(SRCS:.cc=.o)
 MODULE_big = plv8-$(PLV8_VERSION)
 EXTENSION = plv8
 PLV8_DATA = plv8.control plv8--$(PLV8_VERSION).sql
+USE_SYSTEM_V8 = 0
 
 ifeq ($(OS),Windows_NT)
 	# noop for now
@@ -34,6 +35,7 @@ ifeq ($(NUMPROC),0)
 	NUMPROC = 1
 endif
 
+ifeq ($(USE_SYSTEM_V8),0)
 SHLIB_LINK += -Ldeps/v8-cmake/build
 
 all: v8 $(OBJS)
@@ -48,6 +50,7 @@ deps/v8-cmake/build/libv8_libbase.a: deps/v8-cmake/README.md
 	@cd deps/v8-cmake && mkdir -p build && cd build && cmake -Denable-fPIC=ON -DCMAKE_BUILD_TYPE=Release ../ && make -j $(NUMPROC)
 
 v8: deps/v8-cmake/build/libv8_libbase.a
+endif
 
 # enable direct jsonb conversion by default
 CCFLAGS += -DJSONB_DIRECT_CONVERSION
-- 
2.49.0

